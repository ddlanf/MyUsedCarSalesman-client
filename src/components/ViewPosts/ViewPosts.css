.view-posts-link{
  color: black;
  top: 10%;
  text-decoration: none;
  position: relative;
  top: 8vw;
}

.view-posts-post-box{
    width: 100%;
    margin-top: 0;
    margin-bottom: 8%;
    margin-left: auto;
    margin-right: auto;
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    align-items: flex-start;
    align-content: space-around;
    flex-wrap: wrap;
  }
  
  
  .view-posts-create-post-button{
    display: block;
    margin-left: auto;
    margin-right: auto;
    margin-top: 30px;
    font-size: 4vw;
    margin-bottom: 8%;
    background-color: rgba(237, 247, 102, 0.952);
    border-radius: 10%;
    border: 1px solid #3c484d;
    font-weight: bold;
    padding-top: 3%;
    padding-bottom: 3%;
    padding-left: 2%;
    padding-right: 2%;
  }

  .view-posts-create-post-button:hover{
    background-color: #996060;
    transition: 0.5s;
    cursor: pointer;
  }

@media screen and (min-width: 1000px) {
    .view-posts-post-box{
        margin-top: 6%;
    }

    .view-posts-create-post-button{
      font-size: 2vw;
      padding-top: 1%;
      padding-left: 2%;
      padding-bottom: 1%;
    }
}