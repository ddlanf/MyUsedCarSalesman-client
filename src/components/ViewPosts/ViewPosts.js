import React, { Component } from 'react'
import './ViewPosts.css'
import { Link } from 'react-router-dom'
import Posts from './Posts/Posts'

export default class ViewPosts extends Component {
    render() {
        return (
            <>
                <section className="view-posts-post-box">
                    <Posts/>
                </section>
                <button className="create-post-button"> 
                    <Link
                        className="view-posts-link"
                        to="./make-post"
                        >
                        Create Post
                    </Link>
                </button>
            </>
        )
    }
}
