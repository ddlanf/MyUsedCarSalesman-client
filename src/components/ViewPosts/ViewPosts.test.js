import React from 'react';
import ReactDOM from 'react-dom';
import ViewPosts from './ViewPosts';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom'

describe('ViewPosts component', () => {
    
    it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter>
                        <ViewPosts/>
                    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
          .create(
                <BrowserRouter>
                    <ViewPosts/>
                </BrowserRouter>)
          .toJSON();
        expect(tree).toMatchSnapshot();  
      });
});