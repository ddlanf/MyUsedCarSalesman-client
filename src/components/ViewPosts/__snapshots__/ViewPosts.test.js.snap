// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewPosts component renders the UI as expected 1`] = `
Array [
  <a
    className="view-posts-link"
    href="/login"
    onClick={[Function]}
  >
    <button
      className="create-post-button"
    >
      Create Post
    </button>
  </a>,
  <section
    className="view-posts-post-box"
  />,
]
`;
